### PR TITLE
Tournaments: Support replace message

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

**Comment by whalemer** [comment]
(Required for Zarel/pokemon-showdown#5600)

File: js/client-chat-tournament.js [hunk 1]
```diff
@@ -307,7 +307,9 @@
 					if (leaves.length > 0) message.push(arrayToPhrase(leaves) + " left the tournament");
 					this.$lastJoinLeaveMessage.text(message.join("; ") + ".");
 					break;
-
+				case 'replace':
```
Comment (Zarel) [inline js/client-chat-tournament.js]: Looks like the surrounding code double spaces cases.

```suggestion

				case 'replace':
```

File: js/client-chat-tournament.js [hunk 2]
```diff
@@ -307,7 +307,9 @@
...
+					this.room.$chat.append("<div class=\"notice\">" + BattleLog.escapeHTML(data[1]) + " has joined the tournament, replacing " + BattleLog.escapeHTML(data[0]) + ".</div>");
+					break;
```
Comment (Zarel) [inline js/client-chat-tournament.js]:
```suggestion
					break;

```

**Comment by Zarel** [comment]
I'm surprised we've never implemented this in the past.

**Comment by Zarel** [comment]
Do we actually want to keep supporting the tournament message ID system? All other server messages just directly send text to the client.

**Comment by whalemer** [comment]
The only reason I can really see is to make it easier for bots to track state, but I'm not sure if that's worth the maintenance effort.

**Comment by whalemer** [comment]
(also battle starts giving a nice message, and batching join / leave messages)

There is a lot that could be moved out, but maybe it's better to have them all as |tournament messages for consistency?

**Comment by Zarel** [comment]
They should probably be kept in there, yes, for ease of `ignoretournaments` ignoring them.

They could all be `|tournament|message|` or something, though.

**Pull request merged**